src/toICN-after.js

```diff
@@ -54,23 +60,41 @@ else{
 }
 //表示書き換え関係
 chordElms.forEach((e) => {
-  let icn = exports.toICN(""+e.firstChild.nodeValue);
-  let isSharp = false;
-  let isSwap = false;
-  let isBlueChord = false;
-  //シャープ、スワップ、特定のセブンスコード等の条件を満たすかどうかを調べる
-  if(icn!=""){
-    e.firstChild.nodeValue = icn;
-    if(icn.includes("#")){isSharp = true;}
-    if(icn.includes("~")){isSwap = true;}
-    if("1[7],1#[7],4[7],4#[7],2[M7],2#[M7],3[M7],5[M7],5#[M7],6[M7],6#[M7],7[M7]".split(",").includes(icn) || /\[sus4\]|\[aug\]|\[dim\]|\[m7\-5\]$/.test(icn)){
-      isBlueChord = true;
+  if(isKeysOnChordElms && e.classList.contains("key")){
+    if(isAutoKeyDetection){
+      keyMatch = e?e.firstChild.nodeValue.match(/(: |：)([A-G](#|b){0,1})(m{0,1})$/):null;
+      key = keyMatch?sharpify(keyMatch[2]):"";
+      minorSignature = keyMatch?keyMatch[4]:"";
+      let keyNo = scale.indexOf(sharpify(key));
+      if(keyMinorSignature=="m"){keyNo += 3;}
+      if(previousKeyNo != -1){
+        let keyModulationDegree = keyNo - previousKeyNo;
+        if(keyModulationDegree >= 7){keyModulationDegree -= 12;}
+        else if(keyModulationDegree <= -6){keyModulationDegree += 12;}
+        e.firstChild.nodeValue += (" ("+(keyModulationDegree>0?"+":"")+keyModulationDegree+")");
+      }
+      previousKeyNo = keyNo;
+    }
+  }
+  else{
+    let icn = module.exports(""+e.firstChild.nodeValue);
```

Comment:
少しリファクタリングして構造が変わっています。

```suggestion
    let icn = exports.toICN(""+e.firstChild.nodeValue);
```